src/components/sign-up-third/SignUpThirdForm.tsx

```diff
@@ -0,0 +1,142 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import { FilledInput, FormControl, InputLabel, MenuItem, Select } from '@mui/material';
+
+import { useForm } from 'react-hook-form';
+
+import { useLocales } from 'src/locales';
+import { saveAddressData } from 'src/redux/slices/addressSlice';
+import { useAppDispatch, useTypedSelector } from 'src/redux/store';
+import { InferType } from 'yup';
+import { useSignUpThirdCountrySelectOptions } from './select-options';
+
+import { ErrorMessage, NextButton, StyledForm } from './style';
+import { useSignUpThirdSchema } from './validation';
+
+interface SignUpThirdFormProps {
+  onNext: () => void;
+}
+
+export default function SignUpThirdForm({ onNext }: SignUpThirdFormProps): JSX.Element {
+  const { translate } = useLocales();
+  const dispatch = useAppDispatch();
+  const signUpThirdSchema = useSignUpThirdSchema();
+  const { countries } = useSignUpThirdCountrySelectOptions();
+  const initialAddressValues = useTypedSelector((state) => state.address.addressData);
+
+  type FormValues = InferType<typeof signUpThirdSchema>;
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors, isValid },
+  } = useForm<FormValues>({
+    resolver: yupResolver(signUpThirdSchema),
+    mode: 'onBlur',
+    defaultValues: initialAddressValues,
+  });
+
+  const selectedCountry = watch('country');
```

Comment:
send me where you found such an implementation example. I don't want to see any constant in the component. it lead to reinitialization with each rerender

Reply:
fixed